cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -49,12 +49,26 @@ public static void abortQuietly(TransactionSystemClient txClient, Transaction tx
 
   public static void invalidateQuietly(TransactionSystemClient txClient, Transaction tx) {
     try {
-      txClient.invalidate(tx.getWritePointer());
+      if (!txClient.invalidate(tx.getWritePointer())) {
+        LOG.error("Failed to invalidate transaction {}", tx);
+      }
     } catch (Throwable t) {
       LOG.error("Exception when invalidating transaction {}", tx, t);
     }
   }
 
+  /**
+   * Wraps the given {@link Throwable} as a {@link TransactionFailureException} if it is not already an instance of
+   * {@link TransactionFailureException}.
+   */
+  public static TransactionFailureException asTransactionFailure(Throwable t) {
```

Comment:
I think there are two questions here.

1. Does Tephra gives enough information in the exception message telling what when wrong?
On commit(), tephra actually returns a boolean to indicate the result instead of throwing exception, except for the case of `TransactionNotInProgressException` (because it is unexpected).
On invalidate(), tephra also returns a boolean and never throws `TransactionFailureException`, but there can be other runtime exception (e.g. thrift exception).
In both cases, it's the client code (in this case SparkTransactional and SparkTransactionService) to check for the boolean result. The current logic is to throw a new `TransactionFailureException` with the proper message inside (I've enhanced the messages to have more information there). Since a `TransactionFailureException` is thrown, even if this `asTransactionFailure()` is called, the exception won't get wrapped, hence it preserves the original message.

2. When an exception passed to this method is not a `TransactionFailureException`, what message to use for the wrapping exception?
I think we can either have the caller to pass in a message to be used in case if wrapping is needed, or we can simply use the `t.getMessage()` as the message for the wrapping exception. @anew Thoughts?